ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -139,3 +141,206 @@ func TestConstructor_SamePulse(t *testing.T) {
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.UnknownCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is empty",
+			TestRailID: "C5007",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is non-empty",
+			TestRailID: "C5008",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: &payload.VCallResult{ReturnArguments: []byte("123")},
+			},
+		},
+		{
+			Name:       "status is empty and pending is known",
+			TestRailID: "C5012",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+				ProvidedContent:     &payload.VStateReport_ProvidedContentBody{},
+			},
+			VStateSendBefore: true,
+			VDelegatedCall: &payload.VDelegatedCallRequest{
+				Callee:       payload.Reference{},
+				CallOutgoing: payload.Reference{},
+				CallFlags:    payload.BuildCallFlags(isolation.Interference, isolation.State),
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			t.Log(test.TestRailID)
+			if test.Skipped != "" {
+				t.Skip(test.Skipped)
+			}
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class         = gen.UniqueGlobalRef()
+				previousPulse = server.GetPulse().PulseNumber
+				outgoingLocal = gen.UniqueLocalRefWithPulse(previousPulse)
+				outgoing      = reference.NewRecordOf(server.GlobalCaller(), outgoingLocal)
+				object        = outgoing
+			)
+
+			test.VState.Object = object
+			test.VState.OrderedPendingEarliestPulse = previousPulse
+			if test.VState.ProvidedContent != nil {
+				test.VState.ProvidedContent.OrderedQueue = []reference.Global{outgoing}
+			}
+
+			if test.VFindCall != nil {
+				test.VFindCall.Callee = outgoing
+				test.VFindCall.Outgoing = outgoing
+			}
+
+			if test.VDelegatedCall != nil {
+				test.VDelegatedCall.Callee = outgoing
+				test.VDelegatedCall.CallOutgoing = outgoing
+			}
+
+			{
+				requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+				requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+				executionMock := runnerMock.AddExecutionMock(outgoing.String())
+				executionMock.AddStart(nil, &execution.Update{
+					Type:   execution.Done,
+					Result: requestResult,
+				})
+			}
```

Comment:
I don't see differences in reply from execution and resent reply from previous executor, so test doesn't check that in some situations call is not executed. Actually in most cases test shouldn't be executed at all.

Reply:
fixed